docs/data-collect-prep/ingesting_data.md

```diff
@@ -0,0 +1,101 @@
+
+Running the code locally is very useful for easy debugging and development of the code. 
+When the code moves to a stable status, it is usually recommended to run it "remotely" on a pod running in the 
+Kubernetes cluster. This allows setting up specific resources to the processing pod 
+(such as memory, CPU and execution priority).
+
+MLRun provides facilities to create `DataItem` objects as inputs to running code. For example, this is a basic
+data ingestion function:
+
+```python
+def ingest_data(context, source_url: mlrun.DataItem):
+    # Load the data from its source, and convert to a DataFrame
+    df = source_url.as_df()
+    
+    # Perform data cleaning and processing
+    # ...
+    
+    # Save the processed data to the artifact store
+    context.log_dataset('cleaned_data', df=df, format='csv')
+```
+
+This code can be placed in a python file, or as a cell in the Python notebook. For example, if the code above was saved
+to a file, the following code creates an MLRun function from it and executes it remotely in a pod:
+
+```python
+# create a function from py or notebook (ipynb) file, specify the default function handler
+ingest_func = mlrun.code_to_function(name='ingest_data', filename='./ingest_data.py', 
+                                     kind='job', image='mlrun/mlrun')
+
+source_url = "s3://input-data/input_data.csv"
+
+ingest_data_run = ingest_func.run(name='ingest_data',
+                                  handler=ingest_data,
+                                  inputs={'source_url': source_url},
+                                  local=False)
+```
+
+As the `source_url` is part of the function's `inputs`, MLRun automatically wraps it up with a `DataItem`. The output
+is logged to the function's `artifact_path`, and can be obtained from the run result:
+
+```python
+cleaned_data_frame = ingest_data_run.artifact('cleaned_data').as_df()
+```
+
+Note that running the function remotely may require attaching storage to the function, as well as passing storage
+credentials through project secrets. See the following pages for more details:
+
+1. [Attach storage to functions](../runtimes/function-storage.html)
+2. [Working with secrets](../secrets.html)
+
+## Data processing
+Once the data is imported from its source, it can be processed using any framework. MLRun natively supports working
+with Pandas DataFrames and converting from and to its `DataItem` object.
+
+For distributed processing of very large datasets, MLRun integrates with the Spark processing engine, and provides
+facilities for executing pySpark code using a Spark service (which can be deployed by the platform when running MLRun
+as part of an Iguazio system) or through submitting the processing task to Spark-operator. The following pages provide
+additional details and code-samples:
+
+1. [Spark service](???) - **do we have a page for this? Are we documenting it?**
+2. [Spark operator](../runtimes/spark-operator.html)
```

Comment:
- {ref}`spark-operator`

Leave out Spark service for now, it's in the list of TODOs...